src/main/java/org/dependencytrack/persistence/TagQueryManager.java

```diff
@@ -0,0 +1,64 @@
+package org.dependencytrack.persistence;
+
+import alpine.common.logging.Logger;
+import alpine.persistence.PaginatedResult;
+import alpine.resources.AlpineRequest;
+import org.dependencytrack.model.Policy;
+import org.dependencytrack.model.Project;
+import org.dependencytrack.model.Tag;
+
+import javax.jdo.PersistenceManager;
+import javax.jdo.Query;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.HashSet;
+
+final class TagQueryManager extends QueryManager implements IQueryManager {
+
+    private static final Logger LOGGER = Logger.getLogger(ProjectQueryManager.class);
+
+    /**
+     * Constructs a new QueryManager.
+     * @param pm a PersistenceManager object
+     */
+    TagQueryManager(final PersistenceManager pm) {
+        super(pm);
+    }
+
+    /**
+     * Constructs a new QueryManager.
+     * @param pm a PersistenceManager object
+     * @param request an AlpineRequest object
+     */
+    TagQueryManager(final PersistenceManager pm, final AlpineRequest request) {
+        super(pm, request);
+    }
+
+    public PaginatedResult getTags(String policyUuid) {
+
+        LOGGER.debug("Retrieving tags under policy " + policyUuid);
+
+        QueryManager qm = new QueryManager();
+        Policy policy = qm.getObjectByUuid(Policy.class, policyUuid);
```

Comment:
Creating a new `qm` here is not necessary. You can use `this` (or omit the receiver) in the following calls.

```suggestion
        Policy policy = getObjectByUuid(Policy.class, policyUuid);
```